### ISSUE TITLE
Add option to asciidoctor cli to specify a config file

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#566]
Invocations of the `asciidoctor` command can get lengthy and tedious to type out. We should offer the option of passing a YAML (and/or a TomDoc) config file that specifies the options and attributes for the processor. In the future, this may expand to other uses as well.

I propose the following command options:

```
-c , --config-file <filename>
```

Example:

config.yml

```
:backend: docbook
:template_dir: ../asciidoctor-backends/slim
:attributes:
  toc: ''
  idseparator: -
  idprefix: ''
```

Used as:

```
asciidoctor -c config.yml sample.ad
```

The config file would be processed in the Asciidoctor#load method. Ultimately, it gets translated into the options argument passed to the API.

**Comment by mojavelinux** [comment]
This can be accomplished today using a preprocessor. The preprocessor would read the config file, convert the values to a map, assign those values to the document object and return the reader (or null) to proceed.

See https://github.com/asciidoctor/asciidoctor-extensions-lab/blob/master/lib/front-matter-preprocessor/extension.rb for an example of a similar extension. (The main difference is that this preprocessor would never touch the reader since it is getting information from a separate file).

**Comment by atbest** [comment]
So can we do it now?

**Comment by mojavelinux** [comment]
Another idea is to make this an implicit include file that contains shared attributes for a project of documents. Once we have this config file, that might be something that an extension handles, or it could be something that a tool does. But what we'd do in core is define a common name and place for this file and describe how it would be interpreted. In other words, esp a spec.

**Comment by mojavelinux** [comment]
I think we should allow the data to be stored underneath the `asciidoc` key. If present, the value of that key will be used as the config data. This allows the configuration file to contain other things (for example, an Antora playbook file).

Here's an example:

```yaml
asciidoc:
  attributes:
    toc: ''
```

**Comment by mojavelinux** [comment]
We will tolerate either String or Symbol keys for portability.

**Comment by mojavelinux** [comment]
I just submitted a PR. Interested to hear some feedback before we move forward with it.

**Comment by mojavelinux** [comment]
No, we don't want it to be bound to the processor. This is AsciiDoc configuration we are talking about. That is the model we adopted for Antora and we're going to stick with that.

I suppose maybe we could consider looking for either key...but the preferred key is `asciidoc`.
